### PR TITLE
Comp Merge

### LINKED ISSUE #180: Get the Starting Pose for Auto for Placement LEDs

### PR DESCRIPTION
**Pull request by alicekuznetsov**
AUTO TESTING:
-1. Finish adding event markers to the paths (Will be done before the meeting starts. its just intaking)
1. Tune Autonomous pid using a 1m by 1m square
2. Test WING Autos
	- We're mostly worrying about the zeroing of the pitch here
3. Figure out how far down the centerline auto can start
4. Test Centerline autos
5. cry

### CODE REVIEW

**Comment by alicekuznetsov** [comment]
Code_QyoDLmahb6
@Alenguye582  The green ghost is the LEDs desired pose :D
ok I'm going to sleep now

**Comment by alicekuznetsov** [comment]
I will focus on making auto paths sometime tonight if I can. Here's the auto list I think is realistic for phr. Most of the autonomous work has already been done (sim is awesome); its mostly getting vision now and then tuning the auto pid/paths on the real bot. 

We also (and i cannot stress this enough) NEED to test driving on the opposing alliance. I'm almost confident that we need to flip the driver controls in some way.

# WING AUTOS:
**Start at Top of Subwoofer**, score all 4 in a line into speaker
**Start at Bottom of Subwoofer**, score all 4 in a line into speaker

# CENTERLINE "DASH" AUTOS:
**Start in-between the top and bottom of subwoofer**, immediately score in speaker then do a mad dash (top down)
**Start as low down on the field as possible**, immediately score in speaker then do a mad dash (bottom up)

# OTHER AUTOS
**Start at the top of sub**, immediately score in the AMP
**Start at some other location and just shoot into the speaker** (the point will be to be out of the way. where should this be?)

# Heres my elegant visualization of starting positions
auto starting pos

**Comment by alicekuznetsov** [comment]
@TaylerUva @Alenguye582 ^^

**Comment by alicekuznetsov** [comment]
Opening this pr not because it's ready, but because I will forget to open it when it is

**Comment by alicekuznetsov** [comment]
i will clean up this pr before Thursday (it is SO messy rn)

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 1]
```diff
@@ -106,11 +108,29 @@ public void configure() {
     driveConfiguration.Slot0.kI = prefDrivetrain.driveI.getValue();
     driveConfiguration.Slot0.kD = prefDrivetrain.driveD.getValue();
 
+    driveConfiguration.CurrentLimits.SupplyCurrentLimitEnable = true;
+    driveConfiguration.CurrentLimits.SupplyCurrentThreshold = 50;
+    driveConfiguration.CurrentLimits.SupplyCurrentLimit = 40;
+    driveConfiguration.CurrentLimits.SupplyCurrentThreshold = 0.1;
```
Comment (alicekuznetsov) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:114]: dude how did our drivetrain even drive what the

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 2]
```diff
@@ -106,11 +108,29 @@ public void configure() {
...
+
     steerConfiguration.Slot0.kS = prefDrivetrain.steerKs.getValue();
     steerConfiguration.Slot0.kP = prefDrivetrain.steerP.getValue();
     steerConfiguration.Slot0.kI = prefDrivetrain.steerI.getValue();
     steerConfiguration.Slot0.kD = prefDrivetrain.steerD.getValue();
 
+    steerConfiguration.CurrentLimits.SupplyCurrentLimitEnable = true;
+    steerConfiguration.CurrentLimits.SupplyCurrentThreshold = 40;
+    steerConfiguration.CurrentLimits.SupplyCurrentLimit = 30;
+    steerConfiguration.CurrentLimits.SupplyCurrentThreshold = 0.1;
```
Comment (alicekuznetsov) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:128]: im not quite sure what to do with this information

File: src/main/java/frc/robot/subsystems/Drivetrain.java (hunk 1 above)
Comment (alicekuznetsov) [reply src/main/java/frc/robot/subsystems/Drivetrain.java:114]: I was gonna make these into preferences before making a pr but this has set me back

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 3]
```diff
@@ -15,14 +15,21 @@
...
     configure();
   }
 
   public void configure() {
-    rollerMotor.getConfigurator().apply(new TalonFXConfiguration());
+    rollerConfig = new TalonFXConfiguration();
+    rollerConfig.CurrentLimits.SupplyCurrentLimitEnable = true;
+    rollerConfig.CurrentLimits.SupplyCurrentThreshold = 40;
+    rollerConfig.CurrentLimits.SupplyCurrentLimit = 30;
+    rollerConfig.CurrentLimits.SupplyCurrentThreshold = 0.1;
```
Comment (alicekuznetsov) [inline src/main/java/frc/robot/subsystems/Intake.java:30]: ITS EVERYWHERE?!?!?!

File: src/main/java/frc/robot/subsystems/Drivetrain.java (hunk 1 above)
Comment (alicekuznetsov) [reply src/main/java/frc/robot/subsystems/Drivetrain.java:114]: as it stands, current limiting is being applied for any time that the motor draws more than 40 amps.
"Delay supply current limiting until current exceeds this threshold for longer than SupplyTimeThreshold.  This allows current draws above SupplyCurrentLimit for a fixed period of time.  This has no effect if SupplyCurrentLimit is greater than this value."

File: src/main/java/frc/robot/subsystems/Pitch.java [hunk 4]
```diff
@@ -61,6 +62,11 @@ public void configure() {
     pitchConfig.SoftwareLimitSwitch.ReverseSoftLimitEnable = true;
     pitchConfig.SoftwareLimitSwitch.ReverseSoftLimitThreshold = prefPitch.pitchReverseLimit.getValue();
 
+    pitchConfig.CurrentLimits.SupplyCurrentLimitEnable = true;
+    pitchConfig.CurrentLimits.SupplyCurrentThreshold = 50;
+    pitchConfig.CurrentLimits.SupplyCurrentLimit = 30;
+    pitchConfig.CurrentLimits.SupplyCurrentThreshold = 0.1;
```
Comment (alicekuznetsov) [inline src/main/java/frc/robot/subsystems/Pitch.java:68]: DUDE ITS EVERYWHERE 😭

File: src/main/java/frc/robot/subsystems/Pitch.java [hunk 5]
```diff
@@ -201,6 +211,10 @@ public Optional<Rotation2d> getDesiredAngleToLock(Pose2d robotPose, Pose3d[] fie
     return Optional.of(desiredAngle);
   }
 
+  public boolean isPitchLocked() {
```
Comment (alicekuznetsov) [inline src/main/java/frc/robot/subsystems/Pitch.java:214]: This is a band-aid fix for a bug; the desired angle SHOULD be the locking angle. In some cases, it was not. I don't know why and I can't give a complete diagnosis without seeing logs

**Review by alicekuznetsov — COMMENTED** [review]
@TaylerUva Take your time getting to this comment/pr; i am incapable of taking a break 

I glanced over this entire pr for bugs. It has 33 JAVA files and i think i almost died in the process. I left comments and made issues for major things i noticed; otherwise, I think its okay. This pr contains all of our testing from the past week or so so its obviously huge

**Review by TaylerUva — APPROVED** [review]

**Pull request merged**